drafts/npm.md

```diff
@@ -0,0 +1,472 @@
+# npm Best Practices Guide
+
+This document aims to be a one-time stop explaining the security supply-chain
+best practices when using npm's package manager. It is one of the workstreams
+of the OSSF's [Best Practices for Open Source
+Developers](https://github.com/ossf/wg-best-practices-os-developers) working
+group. This document provides 1) an overview of security features of npm in the
+context of supply-chain, 2) explicit recommendations and 3) details or links to
+the official documentation to achieve these recommendations. This document is
+intended to complement the official npm documentation, not to be an
+alternative.
+
+## TOC
+
+- [npm Best Practices Guide](#npm-best-practices-guide)
+  * [TOC](#toc)
+  * [Dependency management](#dependency-management)
+    + [Intake](#intake)
+    + [Declaration](#declaration)
+    + [Reproducible installation](#reproducible-installation)
+      - [Vendoring dependencies](#vendoring-dependencies)
+      - [Hash pinning](#hash-pinning)
+        * [Package-lock.json](#package-lockjson)
+        * [Shrinkwrap.json](#shrinkwrapjson)
+    + [Maintenance](#maintenance)
+  * [Release](#release)
+    + [Account](#account)
+    + [Signing and Verification](#signing-and-verification)
+    + [Publishing](#publishing)
+  * [Private packages](#private-packages)
+    + [Scopes](#scopes)
+    + [Private registry configurations](#private-registry-configurations)
+
+## Dependency management
+
+### Intake
+
+The first step to using a dependency is to study its origin, trustworthiness
+and security posture. Projects like [Envoy
+proxy](https://github.com/envoyproxy/envoy) have [well-documented
+criteria](https://github.com/envoyproxy/envoy/blob/main/DEPENDENCY_POLICY.md#new-external-dependencies)
+a dependency must meet before it is used.
+
+**Recommendations:**
+
+1. **Be aware of typosquatting attacks**, when an attacker creates an
+   official-looking package name to trick users into installing rogue packages
+   ([1](https://threatpost.com/discord-stealing-malware-npm-packages/163265/),
+   [2](https://blog.sonatype.com/damaging-linux-mac-malware-bundled-within-browserify-npm-brandjack-attempt),
+   [3](https://blog.npmjs.org/post/163723642530/crossenv-malware-on-the-npm-registry)). Although
+   the npm registry performs scans to detect typosquatting, no system is
+   perfect, so stay vigilant. Identify the GitHub repository of the package and
+   assess its trustworthiness through, for example, its number of contributors,
+   stars, etc.
+
+    1. Note: Non-ASCII characters and uppercase ASCII characters are [not
```

Comment:
uppercase characters are supported for historical packages, but not for new ones, so it would be incorrect for tooling to ban them entirely.